database/database_test.go

```diff
@@ -177,6 +178,41 @@ func TestStateForDomain(t *testing.T) {
 	}
 }
 
+func TestStatesForDomains(t *testing.T) {
+	resetDB()
+
+	if err := testDB.PutState(DomainState{Name: "a.test", Status: StatusPreloaded}); err != nil {
+		t.Errorf("cannot put state %s for test StatatesForDomains", err)
+		return
+	}
+	if err := testDB.PutState(DomainState{Name: "b.test", Status: StatusPending}); err != nil {
+		t.Errorf("cannot put state %s for test StatesForDomains", err)
+		return
+	}
+
+	domainStates, err := testDB.StatesForDomains([]string{"a.test", "b.test"})
+	if err != nil {
+		t.Errorf("Cannot get states for test StatesForDomains: %s", err)
+		return
+	}
+
+	sort.Slice(domainStates, func(i, j int) bool { return domainStates[i].Name < domainStates[j].Name })
+	if len(domainStates) != 2 {
+		count := strconv.Itoa(len(domainStates))
+		t.Errorf("Inccorect count of states for test StatesForDomains %s", count)
+	}
+	if domainStates[0].Name != "a.test" || domainStates[1].Name != "b.test" {
+		t.Errorf("Domain states not populated correctly for test StatesForDomains")
```

Comment:
Consider adding the got/want information in this error too so a failing test is easier to debug.

Reply:
Done